src/libfetchers/input-accessor.hh

```diff
@@ -0,0 +1,149 @@
+#pragma once
+
+#include "ref.hh"
+#include "types.hh"
+#include "archive.hh"
+#include "canon-path.hh"
+#include "repair-flag.hh"
+
+namespace nix {
+
+class StorePath;
+class Store;
+
+struct InputAccessor
+{
+    enum Type { tRegular, tSymlink, tDirectory, tMisc };
```

Comment:
Is `tMisc` the `"unknown"` type in the language?

```
nix-repl> lib.pathType "/dev/null"
"unknown"
```

Can it be used like `DT_UNKNOWN` to delay the lookup of the type? (I would guess no, but probably good idea to address the phenomenon.)

```suggestion
    enum Type {
      tRegular, tSymlink, tDirectory,
      /**
        Any other node types that may be encountered on the file system, such as device nodes, sockets, named pipe, and possibly even more exotic things.

        Responsible for `"unknown"` from `builtins.readFileType "/dev/null"`.

        Unlike `DT_UNKNOWN`, this must not be used for deferring the lookup of types.
      */
      tMisc
    };
```